### ISSUE TITLE
Sender profile should update as we see more information.

### ISSUE THREAD

**Issue by MatthewChow** [mikeaboody/phishing-research#51]

**Comment by davidwagner** [comment]
This will probably get easier to implement after my cleanups to the detectors.  They bring create_sender_profile() closer to a simple loop where we just do some simple code on each email; that simple code could be factored out to update_sender_profile(), which gives a way to update sender profile as each email is received.  Will still need to do that refactoring and to add the scaffolding in feature generation to call that.

**Comment by davidwagner** [comment]
Incidentally, one challenge here is that we'll need to restructure/reorder the code in generate_data_matrix(), to make sure that we're testing phish emails with the same amount of emails added to the profile as legit emails.  One possible way to do that is something like this:

    for i in num_emails_to_use_for_data_matrix:
        for detector in detectors:
            features_legit = detector.classify(legit_inbox[i])
            features_phish = detector.classify(phish_inbox[i])
            detector.update_sender_profile(legit_inbox[i])
            ....

so that phish emails are classified using the same amount of prior history as legit emails.  We don't want the feature vectors for legit emails to be trained with a variable number of prior history, whereas feature vectors for simulated phish emails to all be trained with the maximum possible amount of prior history.

**Comment by davidwagner** [comment]
See PR #105: please review, and if it looks good, merge.  This is a first step towards addressing this issue.

**Comment by davidwagner** [comment]
See PR #106: please review, and if looks good, merge.  This is the second step towards addressing this issue.  It builds on PR #105 above.  If you just want to see what's different from PR#105, look only at commit 101bdd1.

**Comment by davidwagner** [comment]
I realized that I forgot to add something.  The above PR's call `update_sender_profile()` when generating the training set, but never called it when generating the test set.  I fixed this in commit dd46c7c.
